Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarCloudWebServiceTest.cs

```diff
@@ -138,5 +153,140 @@ public async Task IsServerLicenseValid_AlwaysTrue()
             isValid.Should().BeTrue();
             logger.AssertDebugMessageExists("SonarCloud detected, skipping license check.");
         }
+
+        #region CACHESTUFF
+
+        [TestMethod]
+        public async Task DownloadCache_NullArgument()
+        {
+            (await sut.Invoking(x => x.DownloadCache(null)).Should().ThrowAsync<ArgumentNullException>()).And.ParamName.Should().Be("localSettings");
+        }
+
+        [TestMethod]
+        [DataRow("", "", "", "", "ProjectKey parameter was not provided. Incremental PR analysis is disabled.")]
+        [DataRow("project", "", "", "", "Organization parameter was not provided. Incremental PR analysis is disabled.")]
+        [DataRow("project", "organization", "", "", "Base branch parameter was not provided. Incremental PR analysis is disabled.")]
+        [DataRow("project", "organization", "branch", "", "Token parameter was not provided. Incremental PR analysis is disabled.")]
+        [DataRow("project", "organization", "branch", "token", "CacheBaseUrl was not successfully retrieved. Incremental PR analysis is disabled.")]
+        public async Task DownloadCache_InvalidArguments(string projectKey, string organization, string branch, string token, string infoMessage)
+        {
+            sut = new SonarCloudWebService(MockIDownloader(), uri, version, logger);
+            var localSettings = GetLocalSettings(projectKey, branch, organization, token);
+
+            var res = await sut.DownloadCache(localSettings);
+
+            res.Should().BeEmpty();
+            logger.AssertSingleInfoMessageExists(infoMessage);
+        }
+
+        [TestMethod]
+        [DataRow("http://cacheBaseUrl:222", "http://cacheBaseUrl:222/v1/sensor_cache/prepare_read?organization=org42&project=project-key&branch=project-branch")]
+        [DataRow("http://cacheBaseUrl:222/", "http://cacheBaseUrl:222/v1/sensor_cache/prepare_read?organization=org42&project=project-key&branch=project-branch")]
+        [DataRow("http://cacheBaseUrl:222/sonar/", "http://cacheBaseUrl:222/sonar/v1/sensor_cache/prepare_read?organization=org42&project=project-key&branch=project-branch")]
+        public async Task DownloadCache_RequestUrl(string cacheBaseUrl, string cacheFullUrl)
+        {
+            var token = "42";
```

Comment:
```suggestion
            const string token = "42";
```

Reply:
I moved this to a standalone field, since I reuse it in more than one methods.